plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -62,14 +68,27 @@
 
   // Preference store (should be configuration scoped) from which we get UUID, opt-in status, etc.
   private IEclipsePreferences preferences;
-
-  private OptInDialogCreator optInDialogCreator;
+  private Display display;
+
+  private ConcurrentLinkedQueue<PingEvent> pingEventQueue;
```

Comment:
I'm afraid I can't. `sendPing()` adds an element to the queue, and the methods can be called concurrently.